### PR TITLE
fix(DTFS2-7492): await tfm deal update

### PR DESCRIPTION
**Pull request by MarRobSoftwire**
## Introduction :pencil2:
When submitting a deal from portal to TFM, the deal is not shown in ACBS (on staging). The logs revealed an error that can be traced to not awaiting a deal update

## Resolution :heavy_check_mark:
Await the deal update in tfm-api

### CODE REVIEW

**Review by AlexBramhill — APPROVED** [review]

File: trade-finance-manager-api/src/v1/controllers/deal.submit.controller.js [hunk 1]
```diff
@@ -107,7 +107,7 @@ const submitDealAfterUkefIds = async (dealId, dealType, checker, auditDetails) =
     }
 
     // Update the deal with all the above modifications
-    const tfmDeal = api.updateDeal({ dealId, dealUpdate, auditDetails });
+    const tfmDeal = await api.updateDeal({ dealId, dealUpdate, auditDetails });
```
Comment (abhi-markan) [inline trade-finance-manager-api/src/v1/controllers/deal.submit.controller.js:110]: This is left out on purpose to avoid sending the update deal stage.

**Review by abhi-markan — CHANGES_REQUESTED** [review]

File: trade-finance-manager-api/src/v1/controllers/deal.submit.controller.js (hunk 1 above)
Comment (ttbarnes) [reply trade-finance-manager-api/src/v1/controllers/deal.submit.controller.js:110]: I'm also confused @abhi-markan 

All the api calls in this file are an `await`, quite surprised this was missing

If needs to _not_ await (why? - bizarre) can we add a comment about this please.

Comment (abhi-markan) [reply trade-finance-manager-api/src/v1/controllers/deal.submit.controller.js:110]: Apologies both, I misunderstood this with deal status controller. Good check @MarRobSoftwire `await` should have been there inline with LOC:183.

I tested this and it appears due to an unexpected response from `addPartyUrn` which causes a snowball effect. I would have expected this to fail in last 6 months but above must have developed a race condition.

**Review by abhi-markan — DISMISSED** [review]

File: trade-finance-manager-api/src/v1/controllers/deal.submit.controller.js (hunk 1 above)
Comment (abhi-markan) [reply trade-finance-manager-api/src/v1/controllers/deal.submit.controller.js:110]: @MarRobSoftwire when you get a minute can you please set the base to `2.4.0` so we can regress this on `staging` next so we can get this out on `prod`.

**Comment by MarRobSoftwire** [comment]
Closing PR to point this at 2.4

**Pull request closed without merging**